From caf45ddce0263dea6ffda1bc9793f3b848e4406b Mon Sep 17 00:00:00 2001
From: Anatol Pomozov <anatol.pomozov@gmail.com>
Date: Wed, 28 Sep 2022 08:48:23 -0700
Subject: [PATCH] Use proper arg key for --help option

Running `lpmake --help` leads to a segfault. It happens because invalid key is used for --help option.

Test: build lpmake and make sure --help works
Issue: https://github.com/nmeum/android-tools/issues/73
Change-Id: I8a2c56b6f2c9dc0a3f254e85469ce10d09c02bc7
---
 partition_tools/lpmake.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/partition_tools/lpmake.cc b/partition_tools/lpmake.cc
index e6c4e5305..76372daf0 100644
--- a/partition_tools/lpmake.cc
+++ b/partition_tools/lpmake.cc
@@ -182,7 +182,7 @@ int main(int argc, char* argv[]) {
         { "metadata-slots", required_argument, nullptr, (int)Option::kMetadataSlots },
         { "partition", required_argument, nullptr, (int)Option::kPartition },
         { "output", required_argument, nullptr, (int)Option::kOutput },
-        { "help", no_argument, nullptr, (int)Option::kOutput },
+        { "help", no_argument, nullptr, (int)Option::kHelp },
         { "alignment-offset", required_argument, nullptr, (int)Option::kAlignmentOffset },
         { "alignment", required_argument, nullptr, (int)Option::kAlignment },
         { "sparse", no_argument, nullptr, (int)Option::kSparse },
-- 
2.38.0

